src/tracker/InMemoryTaskManager.java

```
@@ -0,0 +1,210 @@
+
+    public InMemoryTaskManager(HistoryManager historyManager) {
+        this.historyManager = historyManager;
+    }
+
+    @Override
+    public int addNewTask(Task task) {
+        task.setId(nextId++);
+        tasks.put(task.getId(), task);
+        return task.getId();
+    }
+
+    @Override
+    public int addNewEpic(Epic epic) {
+        epic.setId(nextId++);
+        epics.put(epic.getId(), epic);
+        return epic.getId();
+    }
+
+    @Override
+    public int addNewSubtask(Subtask subtask) {
+        subtask.setId(nextId++);
+        subtasks.put(subtask.getId(), subtask);
+        Epic epic = epics.get(subtask.getEpicId());
+        if (epic != null) {
+            epic.addSubtaskId(subtask.getId());
+            updateEpicStatus(epic.getId());
+        }
+        return subtask.getId();
+    }
+
+    @Override
+    public Task getTask(int id) {
+        Task task = tasks.get(id);
+        historyManager.add(task);
+        return task;
+    }
+
+    @Override
+    public Epic getEpic(int id) {
+        Epic epic = epics.get(id);
+        historyManager.add(epic);
+        return epic;
+    }
+
+    @Override
+    public Subtask getSubtask(int id) {
+        Subtask subtask = subtasks.get(id);
+        historyManager.add(subtask);
+        return subtask;
+    }
+
+    @Override
+    public List<Task> getTasks() {
+        return new ArrayList<>(tasks.values());
+    }
+
+    @Override
+    public List<Task> getEpics() {
+        return new ArrayList<>(epics.values());
+    }
+
+    @Override
+    public List<Task> getSubtasks() {
+        return new ArrayList<>(subtasks.values());
+    }
+
+    @Override
+    public Task updateTask(Task task) {
+        if (tasks.containsKey(task.getId())) {
+            tasks.put(task.getId(), task);
+            return task;
+        }
+        return null;
+    }
+
+    @Override
+    public Epic updateEpic(Epic epic) {
+        if (epics.containsKey(epic.getId())) {
```

Весь код метода в if. Можно сделать наоборот, это будет называться fast return или fail fast, то есть в данном случае обратное условие в if и return null; А дальше идет код метода, уже не сдвинутый вправо